core/provider/service.go

```diff
@@ -353,16 +353,41 @@ func (s *Service) Delete(ctx context.Context, id string) error {
 	return nil
 }
 
+func (s *Service) ListAccess(ctx context.Context, providerID string, resources []*domain.Resource) (*domain.Provider, domain.MapResourceAccess, error) {
```

Comment:
This function would just accept `provider type` and the `GetByID` could be done before the caller call this function. No need to have DB dependency for this.